### ISSUE TITLE
Error in demo.py

### ISSUE THREAD

**Issue by shgnag** [Zhongdao/Towards-Realtime-MOT#9]
Thanks for this code. I'm trying to run demo.py by following all the given instructions. I got following two errors. Can you please help me with this?

2019-10-09 11:56:11 [INFO]: start tracking...
Lenth of the video: 4706 frames
2019-10-09 11:56:11 [INFO]: **[Errno 2] No such file or directory: 'weights/latest.pt'**
ffmpeg version 4.2 Copyright (c) 2000-2019 the FFmpeg developers
  built with gcc 7.3.0 (crosstool-NG 1.23.0.449-a04d0)
  configuration: --prefix=******* --cc=/home/conda/feedstock_root/build_artifacts/ffmpeg_1566210161358/_build_env/bin/x86_64-conda_cos6-linux-gnu-cc --disable-doc --disable-openssl --enable-avresample --enable-gnutls --enable-gpl --enable-hardcoded-tables --enable-libfreetype --enable-libopenh264 --enable-libx264 --enable-pic --enable-pthreads --enable-shared --enable-static --enable-version3 --enable-zlib --enable-libmp3lame
  libavutil      56. 31.100 / 56. 31.100
  libavcodec     58. 54.100 / 58. 54.100
  libavformat    58. 29.100 / 58. 29.100
  libavdevice    58.  8.100 / 58.  8.100
  libavfilter     7. 57.100 /  7. 57.100
  libavresample   4.  0.  0 /  4.  0.  0
  libswscale      5.  5.100 /  5.  5.100
  libswresample   3.  5.100 /  3.  5.100
  libpostproc    55.  5.100 / 55.  5.100
[image2 @ 0x55b2a58bdcc0] **Could find no file with path 'results/frame/%05d.jpg' and index in the range 0-4
results/frame/%05d.jpg: No such file or directory**

**Comment by Zhongdao** [comment]
Hi, you should specify the the pre-trained weights by `--weights /path/to/your/weights`.
You can find the download link in README.md.

**Comment by shgnag** [comment]
Thanks for helping with this. Still I got an other issue:

2019-10-09 13:57:33 [INFO]: start tracking...
Lenth of the video: 4706 frames
2019-10-09 13:57:36 [INFO]: Processing frame 0 (100000.00 fps)
2019-10-09 13:57:37 [INFO]: Not compiled with GPU support (nms at /home/Towards-Realtime-MOT-master/maskrcnn_benchmark/maskrcnn_benchmark/csrc/nms.h:22)
frame #0: c10::Error::Error(c10::SourceLocation, std::string const&) + 0x45 (0x7fa6cf77c895 in /home/anaconda3/envs/maskrcnn_benchmark/lib/python3.6/site-packages/torch/lib/libc10.so)
frame #1: nms(at::Tensor const&, at::Tensor const&, float) + 0xdc (0x7fa6b80c949c in /home/Towards-Realtime-MOT-master/maskrcnn_benchmark/maskrcnn_benchmark/_C.cpython-36m-x86_64-linux-gnu.so)
frame #2: <unknown function> + 0x179c8 (0x7fa6b80d79c8 in /home/Towards-Realtime-MOT-master/maskrcnn_benchmark/maskrcnn_benchmark/_C.cpython-36m-x86_64-linux-gnu.so)
frame #3: <unknown function> + 0x14dda (0x7fa6b80d4dda in /home/Towards-Realtime-MOT-master/maskrcnn_benchmark/maskrcnn_benchmark/_C.cpython-36m-x86_64-linux-gnu.so)
<omitting python frames>
frame #32: __libc_start_main + 0xe7 (0x7fa716c9eb97 in /lib/x86_64-linux-gnu/libc.so.6)

ffmpeg version 4.2 Copyright (c) 2000-2019 the FFmpeg developers
  built with gcc 7.3.0 (crosstool-NG 1.23.0.449-a04d0)
  configuration: --prefix=/home/anaconda3/envs/maskrcnn_benchmark --cc=/home/conda/feedstock_root/build_artifacts/ffmpeg_1566210161358/_build_env/bin/x86_64-conda_cos6-linux-gnu-cc --disable-doc --disable-openssl --enable-avresample --enable-gnutls --enable-gpl --enable-hardcoded-tables --enable-libfreetype --enable-libopenh264 --enable-libx264 --enable-pic --enable-pthreads --enable-shared --enable-static --enable-version3 --enable-zlib --enable-libmp3lame
  libavutil      56. 31.100 / 56. 31.100
  libavcodec     58. 54.100 / 58. 54.100
  libavformat    58. 29.100 / 58. 29.100
  libavdevice    58.  8.100 / 58.  8.100
  libavfilter     7. 57.100 /  7. 57.100
  libavresample   4.  0.  0 /  4.  0.  0
  libswscale      5.  5.100 /  5.  5.100
  libswresample   3.  5.100 /  3.  5.100
  libpostproc    55.  5.100 / 55.  5.100
[image2 @ 0x556a0a539cc0] Could find no file with path 'results/frame/%05d.jpg' and index in the range 0-4
results/frame/%05d.jpg: No such file or directory

**Comment by Zhongdao** [comment]
It seems maskrcnn_benchmark is not properly installed, please check their instruction

**Comment by shgnag** [comment]
Thanks for your answer. I have tried many times to install maskrcnn_benchmark as per the given instructions. Still I'm getting this error

ot compiled with GPU support (nms at /home/Towards-Realtime-MOT-master/maskrcnn_benchmark/maskrcnn_benchmark/csrc/nms.h:22)
frame #0: c10::Error::Error(c10::SourceLocation, std::string const&) + 0x45 (0x7fa6cf77c895 in /home/anaconda3/envs/maskrcnn_benchmark/lib/python3.6/site-packages/torch/lib/libc10.so)

**Comment by lilinghui111** [comment]
I have same question.
i run demo in cpu.Is it because of this?

**Comment by Zhongdao** [comment]
@shgnag Did you use the same version of compiler (g++) to compile PyTorch and maskrcnn_benchmark?

**Comment by Zhongdao** [comment]
@lilinghui111 Maybe you used the wrong model, please check issue #8

**Comment by lilinghui111** [comment]
> @shgnag Did you use the same version of compiler (g++) to compile PyTorch and maskrcnn_benchmark?



If I just want to run demo, can I use CPU?

**Comment by Zhongdao** [comment]
@lilinghui111 Yes you can, but you should modify the code slightly, remove all the `.cuda()` command.

**Comment by lilinghui111** [comment]
> Yes you can, but you should modify the code slightly, remove all the `.cuda()` command
yesterday i use windows(cuda 10),
But there is one. so file that cannot be compiled.
Can you give me your VX?

**Comment by lilinghui111** [comment]
> @lilinghui111 Yes you can, but you should modify the code slightly, remove all the `.cuda()` command.

I've seen you say before that you're going to experiment on Windows, and I think I've done almost that on Windows.

**Comment by Zhongdao** [comment]
@lilinghui111 That's great! As for the dynamic library, if you mean `cython_bbox.so`, I think issue #13  will help. You can add me on Wechat (13051902595)  for further discussion.
